src/cache/cloud_fraction.jl

```diff
@@ -33,6 +33,17 @@ NVTX.@annotate function set_cloud_fraction!(
     (; ᶜts, ᶜmixing_length, ᶜcloud_fraction) = p.precomputed
     thermo_params = CAP.thermodynamics_params(params)
     if isnothing(turbconv_model)
+        if p.atmos.call_cloud_diagnostics_per_stage isa
```

Comment:
We need to calculate these gradients as well if set_cloud_fraction is in the callback I think.